### ISSUE TITLE
warn when tests raise plain assertion errors without annotation

### ISSUE THREAD

**Issue by RonnyPfannschmidt** [pytest-dev/pytest#1930]
with the removal of reinterpretation,
we start to expose hard to comprehend issues

for example in #1929 its not clear at first, that the import inside of the conftest causes the rewriting of the test module to fail

in order to protect users we need to make them aware of those subtle issues

**Comment by nicoddemus** [comment]
I agree it would be nice, but I have no idea how to do any kind of checking in this sense...

**Comment by RonnyPfannschmidt** [comment]
when collecting a test module we can check for attribute names starting with `@pytest`

**Comment by nicoddemus** [comment]
But test modules are *always* rewritten... The problem is other modules with asserts that we would like to rewrite too.

**Comment by RonnyPfannschmidt** [comment]
wrong - as far as i can tell from the errors in #1929 test modules do **not** get rewritten when importing them from a conftest first (so the rewritten conftest causes a non-rewritten import of the test module)

**Comment by nicoddemus** [comment]
Oh right... OK, now that makes sense, thanks.

**Comment by obestwalter** [comment]
**tested with pytest==3.6.5 on Python 3.6**

For me this works as I would expect it without any of the workarounds mentioned in #1929 - so has this been fixed in some way or am I missing something?

code used (all modules in the same directory):

`conftest.py`:

```python
from some_code import Foo

def pytest_assertrepr_compare(op, left, right):
    if isinstance(left, Foo) and isinstance(right, Foo) and op == "==":
        return ['Comparing Foo instances:',
                f'\tvalues: "{left.name}" != "{right.name}"']
```

`some_code.py`:

```python
class Foo(object):
    def __init__(self, name):
        self.name = name

    def __eq__(self, other):
        return self.name == other.name

def do_something_with_foo():
    f1 = Foo("just some foo")
    f2 = Foo("another foo")
    return f1, f2
```

`test_some_code.py`:

```python
from some_code import do_something_with_foo

def test_some_code():
    f1, f2 = do_something_with_foo()
    assert f1 == f2
```

ouput:

```text
    def test_some_code():
        f1, f2 = do_something_with_foo()
>       assert f1 == f2
E       assert Comparing Foo instances:
E         	values: "just some foo" != "another foo"

test_some_code.py:6: AssertionError
```

**Comment by RonnyPfannschmidt** [comment]
@obestwalter you never imported the test file in the conftest

i should add an xfailing test

**Comment by obestwalter** [comment]
oh ... the [gist](https://gist.github.com/virhilo/1eb5c1fd952b8b5141c1646a88a51943) demonstrating the problem actually uses a **test module** called `foo2.py` containing the class under test in the same module ... my brain wasn't willing to accept that as reality, so it recalculated the code to what I think it should be like :)

For this (very exotic?) corner case, the problem still persists then.

**Comment by RonnyPfannschmidt** [comment]
@obestwalter its always a bit hilarious how our well meaning "auto-correct" brain mode automatically undoes issues users face that choose interesting or unexpected setups,
